File: chapter-03/exercise-13-determining-compound-interest/DeterminingCompoundInterest.java
import java.math.BigDecimal;
import java.math.RoundingMode;

class DeterminingCompoundInterest {
  public static void main(String[] args) {
    var input = System.console().readLine("What is the principal amount? ");
    var principal = Double.parseDouble(input);

    input = System.console().readLine("What is the rate? ");
    var rate = Double.parseDouble(input);

    input = System.console().readLine("What is the number of years? ");
    var time = Integer.parseInt(input);

    input = System.console().readLine("What is the number times the interest is compounded per year? ");
    var compoundsPerYear = Integer.parseInt(input);

    var amount = new BigDecimal(principal * Math.pow((1 + (rate / 100) / compoundsPerYear), compoundsPerYear * time)).setScale(2, RoundingMode.UP);

    var output = String.format("$%.2f invested at %.2f%% for %d years compounded %d times per year is $%s", principal, rate, time, compoundsPerYear, amount);
    System.out.println(output);
  }
}

File: exercise-13-determining-compound-interest/determining_compound_interest.py
principal = float(input("What is the principal amount? "))
rate = float(input("What is the rate? "))
years = int(input("What is the number of years? "))
compound = int(input("What is the number of times the interest\n"
                     "is compounded per year?"))

amount = round(principal * ((1 + (rate / 100) / compound) ** 
               (compound * years)), 2)

print(f"${principal:.2f} invested at {rate}% for {years} years\n"
      f"compounded {compound} times per year is ${amount:.2f}.")

